src/solution/suite_run.rs

```diff
@@ -0,0 +1,29 @@
+use crate::outputstyle::OutputStyle;
+use super::test_run::TestRun;
+
+pub struct SuiteRun(Vec<TestRun>);
+
+impl SuiteRun {
+    pub fn new(results: Vec<TestRun>) -> Self {
+        Self(results)
+    }
+
+    pub fn results(&self) -> &Vec<TestRun> {
+        &self.0
+    }
+
+    pub fn tests_count(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_successful(&self) -> bool  {
+        self.0.last().expect("At least one test should have been run")
+            .is_successful()
+    }
```

Comment:
This doesn't seem right, it should check that each `TestRun` is successful rather than just the last one.

Reply:
That is true, I had run out of time and was just looking to get it to build xD

Reply:
This should also be resolved now.